x/nexus/keeper/grpc_query.go

```diff
@@ -75,15 +91,15 @@ func (k Keeper) FeeInfo(c context.Context, req *types.FeeInfoRequest) (*types.Fe
 }
 
 // TransferFee returns the transfer fee for a cross chain transfer
-func (k Keeper) TransferFee(c context.Context, req *types.TransferFeeRequest) (*types.TransferFeeResponse, error) {
+func (q Querier) TransferFee(c context.Context, req *types.TransferFeeRequest) (*types.TransferFeeResponse, error) {
 	ctx := sdk.UnwrapSDKContext(c)
 
-	sourceChain, ok := k.GetChain(ctx, req.SourceChain)
+	sourceChain, ok := q.keeper.GetChain(ctx, req.SourceChain)
 	if !ok {
 		return nil, sdkerrors.Wrapf(types.ErrNexus, "%s is not a registered chain", req.SourceChain)
 	}
```

Comment:
```suggestion
        var sourceChain nexus.Chain
	var ok bool

	switch q.axelarnet.IsCosmosChain(ctx, sourceChain.Name) {
	// When source chain is a cosmos chain, use axelarnet for fee info where deposit address is generated on
	case true:
		sourceChain, ok = q.keeper.GetChain(ctx, exported.Axelarnet.Name)
		if !ok {
			return nil, sdkerrors.Wrapf(types.ErrNexus, "%s is not a registered chain", exported.Axelarnet.Name)
		}
	default:
		sourceChain, ok = q.keeper.GetChain(ctx, req.SourceChain)
		if !ok {
			return nil, sdkerrors.Wrapf(types.ErrNexus, "%s is not a registered chain", req.SourceChain)
		}
	}
```
As a general rule of thumb, you wanna have code related to some kind of logic concentrated in one place instead of spread across the whole function.